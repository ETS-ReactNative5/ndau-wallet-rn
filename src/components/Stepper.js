import React, { Component } from 'react';

import { StyleSheet, Image } from 'react-native';

const imageMap = new Map();

class Stepper extends Component {
  constructor(props) {
    super(props);

    imageMap.set(1, require('../../img/step-1.png'));
    imageMap.set(2, require('../../img/step-2.png'));
    imageMap.set(3, require('../../img/step-3.png'));
    imageMap.set(4, require('../../img/step-4.png'));
    imageMap.set(5, require('../../img/step-5.png'));
    imageMap.set(6, require('../../img/step-6.png'));
    imageMap.set(7, require('../../img/step-7.png'));
    imageMap.set(8, require('../../img/step-8.png'));
  }

  render() {
    return (
      <Image
        style={styles.stepper}
        source={imageMap.get(this.props.screenNumber)}
        resizeMode="contain"
        {...this.props}
      />
    );
  }
}

var styles = StyleSheet.create({
  stepper: {
    width: '100%',
    height: 60,
    backgroundColor: 'transparent',
    marginBottom: 5
  }
});

export default Stepper;
