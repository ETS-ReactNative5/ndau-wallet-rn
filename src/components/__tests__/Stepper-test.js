import { mount } from 'enzyme';
import React from 'react';
import { StyleSheet } from 'react-native';
import Stepper from '../Stepper';

import renderer from 'react-test-renderer';

describe('testing Stepper...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
  const navigator = {
    setStyle: () => {}
  };

  beforeEach(() => {});
  it('renders correctly', () => {
    const tree = renderer.create(<Stepper parentStyles={styles} navigator={navigator} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
