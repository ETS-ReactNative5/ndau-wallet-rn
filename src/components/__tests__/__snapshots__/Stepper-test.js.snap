// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Stepper... renders correctly 1`] = `
<Image
  navigator={
    Object {
      "setStyle": [Function],
    }
  }
  parentStyles={
    Object {
      "wizardText": Object {
        "color": "#ffffff",
        "fontSize": 20,
      },
    }
  }
  resizeMode="contain"
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 60,
      "marginBottom": 5,
      "width": "100%",
    }
  }
/>
`;
